### PR TITLE
Proj0243: Generate SBOM

### LINKED ISSUE #240: New Rule: Generate SBOM
To improve transparency, NuGet suggest to include a **Software Bill of Materials** (SBoM) in NuGet packages. This can be enabled by `<GenerateSBOM>true</GenerateSBOM>`.

More info can be found at:
* https://devblogs.microsoft.com/engineering-at-microsoft/microsoft-open-sources-software-bill-of-materials-sbom-generation-tool/
* https://youtu.be/uljRikE6uTU

### PR DESCRIPTION
**Pull request by Corniel**
Closes #240

### CODE REVIEW

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/GenerateSbom.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class GenerateSbom() : MsBuildProjectFileAnalyzer(Rule.GenerateSbom)
+{
+    /// <inheritdoc />
+    public override IReadOnlyCollection<ProjectFileType> ApplicableTo => ProjectFileTypes.ProjectFile;
+
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        if (context.File.IsPackable() && context.File.Property<GenerateSBOM>()?.Value is not true)
```
Comment (CptWesley) [inline src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/GenerateSbom.cs:11]: if `Microsoft.Sbom.Targets` is required, shouldn't we also check for that?

File: src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj [hunk 2]
```diff
@@ -202,6 +204,10 @@ v1.0.0
     <None Include="README.md" Pack="true" PackagePath="" />
   </ItemGroup>
 
+  <ItemGroup Label="Props and targets">
+    <PackageReference Include="Microsoft.Sbom.Targets" Version="3.0.0" PrivateAssets="all" />
```
Comment (CptWesley) [inline src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj:208]: Shouldn't this also be added to the list of packages that should have private assets

Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/DotNetProjectFile.Analyzers.csproj:208]: Yes, it should.

File: src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/GenerateSbom.cs (hunk 1 above)
Comment (Corniel) [reply src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/GenerateSbom.cs:11]: Working on that.

**Review by CptWesley — APPROVED** [review]
LGTM

**Pull request merged**